cfme/common/physical_storage_views.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+from lxml.html import document_fromstring
+from widgetastic_manageiq import (
+    BaseEntitiesView,
+    JSBaseEntity,
+    BreadCrumb,
+    ItemsToolBarViewSelector,
+    SummaryTable,
+    ManageIQTree
+)
+from widgetastic_patternfly import Dropdown, Accordion
+from widgetastic.utils import Parameter
+from widgetastic.widget import ParametrizedView, Text, View
+
+from cfme.base.login import BaseLoggedInPage
+
+
+class ComputePhysicalInfrastructureStoragesView(BaseLoggedInPage):
+    """Common parts for Storage views."""
+    title = Text('.//div[@id="center_div" or @id="main-content"]//h1')
+
+    @property
+    def in_compute_physical_infrastructure_storages(self):
+        return (self.logged_in_as_current_user and
+                self.navigation.currently_selected == ["Compute", "Physical Infrastructure",
+                                                       "Storages"])
+
+
+class PhysicalStorageEntity(JSBaseEntity):
+    @property
+    def data(self):
+        data_dict = super(PhysicalStorageEntity, self).data
+        if 'quadicon' in data_dict and data_dict['quadicon']:
+            quad_data = document_fromstring(data_dict['quadicon'])
+            data_dict['no_port'] = int(quad_data.xpath(self.QUADRANT.format(pos="a"))[0].text)
+            data_dict['state'] = quad_data.xpath(self.QUADRANT.format(pos="b"))[0].get('style')
+            data_dict['vendor'] = quad_data.xpath(self.QUADRANT.format(pos="c"))[0].get('alt')
+            data_dict['health_state'] = quad_data.xpath(self.QUADRANT.format(pos="d"))[0].get('alt')
+        return data_dict
+
+
+class PhysicalStorageDetailsToolbar(View):
+    """Represents physical toolbar and its controls."""
+    configuration = Dropdown(text="Configuration")
+
+    @ParametrizedView.nested
+    class custom_button(ParametrizedView):  # noqa
+        PARAMETERS = ("button_group", )
+        _dropdown = Dropdown(text=Parameter("button_group"))
+
+        def item_select(self, button, handle_alert=False):
+            self._dropdown.item_select(button, handle_alert=handle_alert)
+
+
+class PhysicalStorageDetailsEntities(View):
+    """Represents Details page."""
+    properties = SummaryTable(title="Properties")
+    management_networks = SummaryTable(title="Management Networks")
+    relationships = SummaryTable(title="Relationships")
+    power_management = SummaryTable(title="Power Management")
+    firmwares = SummaryTable(title="Firmwares")
+    ports = SummaryTable(title="Ports")
+
+
+class PhysicalStorageDetailsView(ComputePhysicalInfrastructureStoragesView):
+    """Main PhysicalStorage details page."""
+    breadcrumb = BreadCrumb(locator='.//ol[@class="breadcrumb"]')
```

Comment:
Required:
Already this locator set for you in `wt.pfy` as `ROOT` please refer this
https://github.com/RedHatQE/widgetastic.patternfly/blob/master/src/widgetastic_patternfly/__init__.py#L1821

No need to pass locator